### ISSUE TITLE
Win32 font detection prints exception traceback for strange 1-byte font name

### ISSUE THREAD

**Issue by pushfoo** [pyglet/pyglet#622]
_I do not have a Windows system to trace the issue further. All information has been obtained from @novialriptide_

## Issue Summary
Font detection produces strange exception output under Win32:
```
Exception ignored on calling ctypes callback function: <function _enum_font_names at 0x000001C4183DC940>
Traceback (most recent call last):
  File "C:\users\novia\documents\github\pyglet\pyglet\font\win32query.py", line 344, in _enum_font_names
    name = lf.lfFaceName.decode('utf-8')
UnicodeDecodeError: 'utf-8' codec can't decode byte 0xf8 in position 0: invalid start byte
bytes: <__main__.LOGFONT object at 0x000001C4183F25C0>
Exception ignored on calling ctypes callback function: <function _enum_font_names at 0x000001C4183DC940>
Traceback (most recent call last):
  File "C:\users\novia\documents\github\pyglet\pyglet\font\win32query.py", line 344, in _enum_font_names
    name = lf.lfFaceName.decode('utf-8')
UnicodeDecodeError: 'utf-8' codec can't decode byte 0xf8 in position 0: invalid start byte
bytes: <__main__.LOGFONT object at 0x000001C4183F25C0>
Exception ignored on calling ctypes callback function: <function _enum_font_names at 0x000001C4183DC940>
Traceback (most recent call last):
  File "C:\users\novia\documents\github\pyglet\pyglet\font\win32query.py", line 344, in _enum_font_names
    name = lf.lfFaceName.decode('utf-8')
UnicodeDecodeError: 'utf-8' codec can't decode byte 0xf8 in position 0: invalid start byte
bytes: <__main__.LOGFONT object at 0x000001C4183F25C0>
Exception ignored on calling ctypes callback function: <function _enum_font_names at 0x000001C4183DC940>
Traceback (most recent call last):
  File "C:\users\novia\documents\github\pyglet\pyglet\font\win32query.py", line 344, in _enum_font_names
    name = lf.lfFaceName.decode('utf-8')
UnicodeDecodeError: 'utf-8' codec can't decode byte 0xf8 in position 0: invalid start byte
```
The font triggering the exceptions appears to report `b'\xf8'` as its `lfFaceName` value, with 0 as its `lfCharSet` value. I don't know if this is a corrupt font or what.
 
This information was obtained through [a modified version of `pyglet/fonts/win32query.py` that prints lfFaceName and lfCharSet](https://gist.github.com/pushfoo/2d7101111f31c39b6bfb72ecf9ca4483). The full output can be read here: https://pastebin.com/tBSapDgp

#### Pyglet versions tried:

* `2.0.dev16` via pip as a dependency of Arcade
* 13709d477 cloned from git (tip of master at time of writing)

#### Modified system info output from Arcade:
```
GPU: GeForce RTX 2070/PCIe/SSE2
Open GL Version: (4, 6)
Python: 3.9.0 (tags/v3.9.0:9cf6752, Oct  5 2020, 15:34:40) [MSC v.1927 64 bit (AMD64)]
Platform: win32
```

#### Font folder contents:
Some of the included fonts include non-english symbols.
![image](https://user-images.githubusercontent.com/36696816/171336544-9b8ea0b0-be96-48db-84e5-fe6d7a1fa3ab.png)
There isn't anything that stands out as obviously broken inside that folder.

### Steps To Replicate
1. Use a system similar to the above
2. Run anything that calls font detection, including `python -m pyglet.font.win32query`
3. Errors are produced, but do not crash the program

**Comment by benmoran56** [comment]
Thanks for the detailed information, and for the debug output. 
The GDI font renderer was recently updated to handle utf8 font names. Looking at the raw bytes, it seems that the font is either corrupted (or doesn't have a utf8 name).  Either way, the GDI font renderer probably won't be able to support it. 

If it's possible to share the actual font file we can have a look, but I'm thinking it's probably best for pyglet to suppress the traceback and skip the font. The GDI font backend is quite old, and support is limited compared to the DirectWrite backend.

**Comment by pushfoo** [comment]
I had a hunch that it might be the cs go font, but the original reporting user had to head off for the night. We might be able to narrow it down to an exact file tomorrow. Can you suggest any utilities or any other guidance for doing so? 

> I'm thinking it's probably best for pyglet to suppress the traceback and skip the font.

I agree, with the reservation that it seems reasonable to warn users that a font may be corrupt. I haven't thoroughly read the relevant Win32 API doc, so I don't know if there's a clean way to report the source filename.

**Comment by caffeinepills** [comment]
The fix for utf fonts was merged to 1.5 but not master. We could port this to master, but no guarantee it will fix it. Worth fixing in master though.

**Comment by benmoran56** [comment]
:open_mouth:  I didn't realize that we hadn't already merged those. 
I just cherry-picked the changes, and pushed out a `2.0.dev17` release. 
@pushfoo That should be easier to test than 1.5

**Comment by pushfoo** [comment]
@benmoran56 Ty! However, according to the user, the tip of master / 2.0.dev17 release appears to still print the error ([full output](https://www.toptal.com/developers/hastebin/oganekepig.properties)) when running `python -m pyglet.font.win32query`.

1.5 branch does not print any exceptions or stack trace, and reports multiple fonts named "ø" at the end out of the [output](https://www.toptal.com/developers/hastebin/abokowuhaw.properties). This is consistent with the data displayed in the stack trace, which appears to be a truncated version of [unicode code point (U+00F8)](https://codepoints.net/U+00F8) .

**Comment by caffeinepills** [comment]
This should be resolved in 2.0dev18? Can you verify so we can close? Thanks

**Comment by pushfoo** [comment]
Confirmed as fixed with the most recent dev release, ty!:
> novial — Today at 12:01 AM
> yea
>no errors
